### PR TITLE
[CoreGraphics] Xcode 14 Beta 1-5

### PR DESCRIPTION
**Pull request by chamons**
**Note:** This PR is almost entirely of manual (but simple) bindings, and the documentation of these CG methods is non-existent. I had to add add MarshalAs and change pointers to IntPtr by hand, so please review carefully. 

I wrote "did not crash/return null" manual tests, which I could improve upon if desired, but I'd have to re-learn some matrix math and figure out what each method does under the hood (Alex thought this would likely be fine).

A few important notes:

- CGAffineTransformComponents is in CoreFoundation as it was defined (but not used) in those API diffs. There is a define `CF_DEFINES_CGAFFINETRANSFORMCOMPONENTS` which allows CoreGraphics to declare it in the headers, but I'm assuming that is a hack for Apple.
- The headers/docs for kCGColorSpaceITUR_709_HLG lie and claim it's in older OS than it seems to be, so I assumed latest instead.

### CODE REVIEW

File: src/CoreGraphics/CGPDFScanner.cs [hunk 1]
```diff
@@ -230,5 +230,30 @@ public bool TryPop (out CGPDFStream? value)
...
+#if NET
+		[SupportedOSPlatform ("ios16.0")]
+		[SupportedOSPlatform ("maccatalyst16.0")]
+		[SupportedOSPlatform ("macos13.0")]
+		[SupportedOSPlatform ("tvos16.0")]
+#else
+		[Mac (13,0), iOS (16,0), TV (16,0), MacCatalyst (16,0)]
+#endif
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
```
Comment (rolfbjarne) [inline src/CoreGraphics/CGPDFScanner.cs:243]: This method doesn't return a value, so this attribute seems unnecessary.

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/CoreGraphics/CGPDFScanner.cs (hunk 1 above)
Comment (chamons) [reply src/CoreGraphics/CGPDFScanner.cs:243]: Copy paste error.

File: src/CoreGraphics/CGPath.cs [hunk 2]
```diff
@@ -529,6 +532,170 @@ public void Apply (ApplierFunction func)
...
+		[SupportedOSPlatform ("macos13.0")]
+		[SupportedOSPlatform ("tvos16.0")]
+#else
+		[Mac (13,0), iOS (16,0), TV (16,0), MacCatalyst (16,0)]
+#endif
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
+		static extern bool CGPathIntersectsPath (IntPtr path1, IntPtr path2, [MarshalAs (UnmanagedType.I1)] bool evenOddFillRule);
+
+		public bool DoesIntersect (CGPath? maskPath, bool evenOddFillRule)
```
[the remark this answers is not in the document]
Comment (chamons) [reply src/CoreGraphics/CGPath.cs:694]: I thought above Intersects, but with `CreateLineByIntersectingPath` I was concerned the names would be confusing. This was you have a Does or a Create prefix to make checking/making clear.

I am not sold on this and could be convinced however.

File: tests/monotouch-test/CoreGraphics/AffineTransformTest.cs [hunk 3]
```diff
@@ -499,6 +500,28 @@ public void Invert ()
...
+		[Test]
+		public void MakeWithComponents ()
+		{
+			var components = new CGAffineTransformComponents () {
+				Scale = new CGSize (0.0, 0.0),
+				HorizontalShear = (nfloat)0.0,
+				Rotation = (nfloat)0.0,
+				Translation = new CGVector ((nfloat)1, (nfloat)1),
+			};
+			Assert.NotNull (CGAffineTransform.MakeWithComponents (components));
```
[the remark this answers is not in the document]
Comment (chamons) [reply tests/monotouch-test/CoreGraphics/AffineTransformTest.cs]: Fixed.

File: src/CoreGraphics/CGPDFScanner.cs (hunk 1 above)
Comment (chamons) [reply src/CoreGraphics/CGPDFScanner.cs]: Fixed.

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

File: src/CoreFoundation/CGAffineTransformComponents.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+using CoreGraphics;
+
+namespace CoreFoundation {
+#if NET
+	[SupportedOSPlatform ("ios")]
+	[SupportedOSPlatform ("maccatalyst")]
+	[SupportedOSPlatform ("macos")]
+	[SupportedOSPlatform ("tvos")]
+#endif
```
Comment (tj-devel709) [inline src/CoreFoundation/CGAffineTransformComponents.cs:17]: Is this struct only available in NET?

File: src/CoreGraphics/CGPDFScanner.cs [hunk 5]
```diff
@@ -230,5 +230,29 @@ public bool TryPop (out CGPDFStream? value)
...
+		[SupportedOSPlatform ("ios16.0")]
+		[SupportedOSPlatform ("maccatalyst16.0")]
+		[SupportedOSPlatform ("macos13.0")]
+		[SupportedOSPlatform ("tvos16.0")]
+#else
+		[Mac (13,0), iOS (16,0), TV (16,0), MacCatalyst (16,0)]
+#endif
+		public void Stop ()
+		{
+			CGPDFScannerStop (Handle);
```
Comment (tj-devel709) [inline src/CoreGraphics/CGPDFScanner.cs:255]: Just curious, I see in other methods as well that there is a "Handle" var, where is it coming from?

**Review by tj-devel709 — APPROVED** [review]
Left a few questions, but LGTM

File: src/CoreGraphics/CGPDFScanner.cs (hunk 5 above)
Comment (chamons) [reply src/CoreGraphics/CGPDFScanner.cs:255]: [NSObject.Handle](https://github.com/xamarin/xamarin-macios/blob/main/src/Foundation/NSObject2.cs#L98)

File: src/CoreFoundation/CGAffineTransformComponents.cs (hunk 4 above)
Comment (chamons) [reply src/CoreFoundation/CGAffineTransformComponents.cs:17]: No, but we don't add attributes to structs in legacy usually. We have a test that insists we do in NET6.

**Review by mandel-macaque — APPROVED** [review]

**Comment by chamons** [comment]
Looking at test failures here in xammac tests/monotouch test.

**Comment by chamons** [comment]
Test failures are watch only.

**Pull request merged**